File: BIND_core/src/+fr/+lescot/+bind/+data/MetaSituation.m
%{
Class:

<MetaBase> extension with the specific properties of the situations (none at
the moment).

%}
classdef MetaSituation < fr.lescot.bind.data.MetaBase

    methods
        
        %{
        Function:
        Instanciates a new MetaSituation object, and set the frameworkVariables
        to the correct values.
        
        Returns:
        A MetaBase instance
        %}
        function out = MetaSituation()
            out@fr.lescot.bind.data.MetaBase;
            timeCodeStart = fr.lescot.bind.data.MetaEventVariable();
            timeCodeStart.setName('startTimecode');
            timeCodeStart.setType(fr.lescot.bind.data.MetaDataVariable.TYPE_REAL);
            timeCodeEnd = fr.lescot.bind.data.MetaEventVariable();
            timeCodeEnd.setName('endTimecode');
            timeCodeEnd.setType(fr.lescot.bind.data.MetaDataVariable.TYPE_REAL);
            out.setFrameworkVariables({timeCodeStart timeCodeEnd});
        end
        
        %{
        Function:
        Returns a hash of the object, which is an identifier string which
        is equal between two objects only if the two objects are equal
        (equality being here an esuality of value, not of reference).
        
        Arguments:
        this - The object on which the function is called, optionnal.
        
        Returns:
        A string   
        %}     
        function out = hash(this)
            baseClassHash = hash@fr.lescot.bind.data.MetaBase(this);
            out = [baseClassHash '|' 'SITUATION'];
        end
        
    end
    
end

File: pynd/pynd/data/meta_situation.py
from pynd.data import MetaBase, MetaEventVariable, MetaVariableBaseType


class MetaSituation(MetaBase):

    def __init__(self):
        super().__init__()
        start_timecode = MetaEventVariable()
        start_timecode.set_name('startTimecode')
        start_timecode.set_type(MetaVariableBaseType.REAL)
        end_timecode = MetaEventVariable()
        end_timecode.set_name('endTimecode')
        end_timecode.set_type(MetaVariableBaseType.REAL)
        self._set_framework_variables([start_timecode, end_timecode])

    def hash(self) -> str:
        return super(MetaSituation, self).hash() + '|' + 'SITUATION'
